src/reactapp/src/components/address/hooks/useCountryState.js

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useMemo } from 'react';
+import _get from 'lodash.get';
+import { useFormikContext } from 'formik';
+
+import useAppContext from '../../../hook/useAppContext';
+import { prepareCountryOptions, prepareCountryStateOptions } from '../utility';
+
+export default function useCountryState({ fields }) {
+  const { values, setFieldValue } = useFormikContext();
+  const [{ countryList, stateList }] = useAppContext();
+  const countrySelected = _get(values, fields.country);
+  const regionField = fields.region;
+  const stateSelected = _get(values, regionField);
+
+  // whenever the country is swiched, we need to clear the state input
+  useEffect(() => {
+    if (countrySelected) {
+      const stateListContainsStateSelected =
+        !stateList.length ||
+        _get(stateList, countrySelected, []).find(
+          s => s.code === stateSelected
```

Comment:
As usual, I'm here to say I don't see added value in shortening the variable `state` to `s`. Webpack will rewrite it to a single character anyway, so it costs us nothing to make it more readable with a full variable name 😄

Reply:
done 👍